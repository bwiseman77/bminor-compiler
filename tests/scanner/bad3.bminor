int main() {
    int b = 100;
    a = &b;
    char *c = "\";
    struct d {
        thing1;
    }

    &d->thing = 'a very bad thing';
}





