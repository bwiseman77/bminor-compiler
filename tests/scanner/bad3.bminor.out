-------------------------------------------
           token type  text
-------------------------------------------
               IDENT  int
               IDENT  main
        LEFT_PARENTH  (
       RIGHT_PARENTH  )
        LEFT_BRACKET  {
               IDENT  int
               IDENT  b
              ASSIGN  =
     INTEGER_LITERAL  100
           SEMICOLON  ;
               IDENT  a
              ASSIGN  =
          SCAN ERROR  &
               IDENT  b
           SEMICOLON  ;
                CHAR  char
      MULTIPLICATION  *
               IDENT  c
              ASSIGN  =
      STRING_LITERAL  "
           SEMICOLON  ;
               IDENT  struct
               IDENT  d
        LEFT_BRACKET  {
               IDENT  thing1
           SEMICOLON  ;
       RIGHT_BRACKET  }
          SCAN ERROR  &
               IDENT  d
                 SUB  -
                  GT  >
               IDENT  thing
              ASSIGN  =
          SCAN ERROR  '
               IDENT  a
               IDENT  very
               IDENT  bad
               IDENT  thing
          SCAN ERROR  '
           SEMICOLON  ;
       RIGHT_BRACKET  }
